MetaData/python/JobConfig.py

```diff
@@ -268,7 +278,11 @@ def customize(self,process):
                                 samplepu = self.pu_distribs[matches[0]]
                             puObj.puReWeight = True
                             puObj.puBins = cms.vdouble( map(float, samplepu.probFunctionVariable) )
-                            puObj.mcPu   = samplepu.probValue
+                            #Using input MC PU or by default the probvalues from cfi file for PU reweighting
+                            if pusimu:
+                                puObj.mcPu   =  cms.vdouble(pusimu)
+                            else:
+                                puObj.mcPu   = samplepu.probValue
                             puObj.dataPu = cms.vdouble(putarget)
                             puObj.useTruePu = cms.bool(True)
```

Comment:
As discussed on the egroup, this line need also to be changed to use the "obseved PU".

@sethzenz, looking at the code, I realized that the variable useTruePu is not initialized in C++ if not specified in python. 
 https://github.com/cms-analysis/flashgg/blob/master/MicroAOD/src/GlobalVariablesComputer.cc#L65
Given that the python always set the flag, we are safe, but it would be good to initialize the flag in the constructor.